test/lib/stack-client/stack-client.logout.spec.js

```diff
@@ -0,0 +1,62 @@
+import stack from 'lib/stack-client'
+
+import internal from 'lib/stack-internal'
+
+let internalLogout
+
+describe("stack client", () => {
+
+  describe("logout", () => {
+
+    const stackClient = {
+      token: { token: "mytoken"},
+      uri: "https://test.mycozy.cloud",
+      fetch: jest.fn( 
+        (method, path, body, options) => Promise.resolve(() => {
+          return { status: 200 }
+        })
+      )
+    }
+    
+    const cozyClient = { 
+      getStackClient: () => stackClient
+    }
+
+    const params = {
+      cozyClient,
+      onCreateApp: function() {},
+      onDeleteApp: function() {},
+    }
+
+    beforeAll(() => {
+      internalLogout = internal.logout
+      internal.logout = jest.fn( () => Promise.resolve() )
+      stack.init(params)
+      stack.logout()
+    })
+
+    afterAll(() => {
+      internal.logout = internalLogout
+    })
+
+    it("should not forward to the old internal client", () => {
+      expect( internal.logout ).not.toHaveBeenCalled()
+    })
+
+    it("should have called cozy-client", () => {
+      expect( cozyClient.getStackClient().fetch ).toHaveBeenCalled()
+    })
+
+    it("should not throw", async () => {
+      expect( () => stack.logout() ).not.toThrow()
+      const thrown = await stack.logout().then(
```

Comment:
Maybe you can use the `await expect(stack.logout()).rejects.toBe(undefined)` construct ?

Reply:
hum… I did not know about `resolves` and `rejects`. `resolves` seems the one that I should use (it will fail if the promise rejects)